server/fileserver.go

```diff
@@ -125,6 +115,33 @@ func getUploadHandler(fsPath string) http.Handler {
 				totalBytesWritten += written
 			}
 		}
+		// check if the file is tar.gz that supposed to be uploaded as a folder.
+		// check with query params.
+		isFolder := req.URL.Query().Get("isFolder")
+		if isFolder == "true" && len(uploadedFileNames) == 1{
```

Comment:
what happens if isFolder is true, but there are more than 1 files in the form?

Reply:
and what happens if the is folder is true and there's a single file in the form but it is not a valid tar.gz archive? will the error returned below by Extract indicate about it?